app/models/chargeback_rate_detail.rb

```diff
@@ -1,19 +1,44 @@
 class ChargebackRateDetail < ActiveRecord::Base
   belongs_to :chargeback_rate
   belongs_to :detail_measure, :class_name => "ChargebackRateDetailMeasure", :foreign_key => :chargeback_rate_detail_measure_id
-
-  validates :rate, :numericality => true
+  has_many :chargeback_tiers, :dependent => :destroy
   validates :group, :source, :presence => true
+  validate :complete_tiers, :if => "!respond_to?(:rate)"
+
+  # Set the rates according to the tiers
+  def find_rate(value)
+    @fixed_rate = 0.0
+    @variable_rate = 0.0
+    chargeback_tiers.each do |tier|
+      next if value < tier.start
+      next if value >= tier.end
+      @fixed_rate = tier.fixed_rate
+      @variable_rate = tier.variable_rate
```

Comment:
Can we return the found fixed_rate and variable_rate instead of modifying instance variables?

It's not obvious below that a "find_*" method is modifying state.